internal/controller/nodemaintenancescheduler_controller.go

```diff
@@ -158,8 +158,8 @@ func (r *NodeMaintenanceSchedulerReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
-	if schedCtx.AvailableSlots == 0 {
-		r.Log.Info("no slots available for maintenance scheduling")
+	if len(schedCtx.CandidateMaintenance) == 0 {
```

Comment:
reason for change in order:

its better to see this in logs than the fact that there are no slots available, in case we dont have any candidate note maintenances AND no available slots.